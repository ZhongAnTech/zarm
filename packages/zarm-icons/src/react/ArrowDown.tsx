import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
