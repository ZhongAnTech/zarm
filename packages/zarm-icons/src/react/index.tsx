export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
