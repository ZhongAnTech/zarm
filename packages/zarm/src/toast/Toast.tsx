import React, { useCallback, useEffect, useRef, useState } from 'react';
import classNames from 'classnames';
import { ConfigContext } from '../config-provider';
import Popup from '../popup';
import type { BaseToastProps } from './interface';
import type { HTMLProps } from '../utils/utilityTypes';

export type ToastProps = BaseToastProps & HTMLProps;

const Toast = React.forwardRef<HTMLDivElement, ToastProps>((props, ref) => {
  const { prefixCls: globalPrefixCls } = React.useContext(ConfigContext);
  const prefixCls = `${globalPrefixCls}-toast`;

  const toastRef = ref || React.createRef<HTMLDivElement>();
  const {
    className,
    stayTime,
    content,
    visible: propVisible,
    afterClose,
    style,
    ...others
  } = props;
  const [visible, setVisible] = useState(propVisible!);
  const timerRef = useRef(0);

  const hide = () => {
    setVisible(false);
  };

  const autoClose = useCallback(() => {
    if (stayTime! > 0) {
      timerRef.current = window.setTimeout(() => {
        hide();
        window.clearTimeout(timerRef.current!);
      }, stayTime);
    }
  }, [stayTime]);

  useEffect(() => {
    if (visible) {
      autoClose();
    }
    return () => {
      window.clearTimeout(timerRef.current!);
    };
  }, [autoClose, visible]);

  useEffect(() => {
    setVisible(propVisible!);
  }, [propVisible]);

  return (
    <Popup
      direction="center"
      maskColor="transparent"
      width="70%"
      lockScroll={false}
      {...others}
      visible={visible}
      afterClose={afterClose}
    >
      <div className={classNames(prefixCls, className)} ref={toastRef} style={style}>
        <div className={`${prefixCls}__container`}>{content}</div>
      </div>
    </Popup>
  );
});

Toast.displayName = 'Toast';

Toast.defaultProps = {
  visible: false,
  stayTime: 3000,
  mask: false,
};

export default Toast;
