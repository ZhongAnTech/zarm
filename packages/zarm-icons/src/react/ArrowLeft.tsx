import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
