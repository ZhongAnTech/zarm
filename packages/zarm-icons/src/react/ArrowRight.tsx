import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
