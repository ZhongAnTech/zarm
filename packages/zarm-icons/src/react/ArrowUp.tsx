import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowUp = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M513.176 285.05a41.472 41.472 0 0120.024 11.127L867.11 630.089c16.272 16.271 16.272 42.653 0 58.925s-42.654 16.272-58.925 0L503.612 384.44 199.085 688.967c-16.272 16.271-42.654 16.271-58.925 0-16.272-16.272-16.272-42.654 0-58.926L474.07 296.129c10.606-10.606 25.508-14.299 39.105-11.078z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowUp);
export default ForwardRef;
