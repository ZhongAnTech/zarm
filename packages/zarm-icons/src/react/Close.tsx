import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgClose = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M560.149 501.223l255.344 255.344c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L501.223 560.149 245.88 815.493c-16.272 16.272-42.654 16.272-58.925 0-16.272-16.272-16.272-42.654 0-58.926l255.344-255.344L186.954 245.88c-16.272-16.272-16.272-42.654 0-58.925 16.271-16.272 42.653-16.272 58.925 0l255.344 255.344 255.344-255.344c16.272-16.272 42.654-16.272 58.926 0 16.272 16.271 16.272 42.653 0 58.925L560.149 501.223z"
          fillRule="nonzero"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgClose);
export default ForwardRef;
