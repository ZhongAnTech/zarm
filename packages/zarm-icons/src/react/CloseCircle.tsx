import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgCloseCircle = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <g fillRule="evenodd">
          <g>
            <path d="M500 460.716l137.493-137.493c10.848-10.848 28.436-10.848 39.284 0l.101.102c10.735 10.848 10.645 28.344-.203 39.08l-138.207 136.78 138.309 138.308c10.848 10.848 10.848 28.436 0 39.284-10.848 10.848-28.436 10.848-39.284 0L498.98 538.264 359.85 675.956c-10.41 10.303-27.188 10.26-37.545-.097-10.34-10.341-10.34-27.107 0-37.448L460.716 500 323.223 362.507c-10.848-10.848-10.848-28.436 0-39.284 10.848-10.848 28.436-10.848 39.284 0L500 460.716z" />
            <path
              d="M190.64 809.36c170.855 170.854 447.865 170.854 618.72 0 170.854-170.855 170.854-447.865 0-618.72-170.855-170.854-447.865-170.854-618.72 0-170.854 170.855-170.854 447.865 0 618.72zm-44.193 44.193c-195.263-195.262-195.263-511.844 0-707.106 195.262-195.263 511.844-195.263 707.106 0 195.263 195.262 195.263 511.844 0 707.106-195.262 195.263-511.844 195.263-707.106 0z"
              fillRule="nonzero"
            />
          </g>
        </g>
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgCloseCircle);
export default ForwardRef;
