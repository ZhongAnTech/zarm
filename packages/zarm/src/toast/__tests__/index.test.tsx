import React from 'react';
import { fireEvent, render, screen, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import Toast from '../index';

jest.useFakeTimers();

describe('Toast', () => {
  test('should renders correctly', () => {
    const { asFragment, rerender } = render(<Toast visible content="foo" />);
    expect(asFragment().firstChild).toMatchSnapshot();
    rerender(<Toast visible={false} content="foo" />);
  });

  test('should show toast when call static method on Toast component', () => {
    const Demo = () => {
      const toast = Toast.useToast();
      return (
        <>
          <button onClick={() => toast.show('toast content')}>open</button>
          <button onClick={() => toast.hide()}>close</button>
        </>
      );
    };
    render(<Demo />);
    fireEvent.click(screen.getByRole('button', { name: 'open' }));
    expect(screen.getByText('toast content')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'close' }));
    expect(screen.queryByText('toast content')).not.toBeInTheDocument();
  });

  test('.show() static method should accepts toast props', () => {
    const Demo = () => {
      const toast = Toast.useToast();
      return (
        <>
          <button onClick={() => toast.show({ content: 'toast content' })}>open</button>
          <button onClick={() => toast.hide()}>close</button>
        </>
      );
    };
    render(<Demo />);
    fireEvent.click(screen.getByRole('button', { name: 'open' }));
    expect(screen.getByText('toast content')).toBeInTheDocument();
    fireEvent.click(screen.getByRole('button', { name: 'close' }));
    expect(screen.queryByText('toast content')).not.toBeInTheDocument();
  });

  test.skip('should close the Toast after 3 seconds by default', () => {
    render(<Toast visible content="foo" />);
    expect(screen.getByText('foo')).toBeInTheDocument();
    jest.advanceTimersByTime(2_000);
    expect(screen.getByText('foo')).toBeInTheDocument();
    act(() => {
      jest.advanceTimersByTime(1_000);
    });
    expect(screen.queryByText('foo')).not.toBeInTheDocument();
  });
});
