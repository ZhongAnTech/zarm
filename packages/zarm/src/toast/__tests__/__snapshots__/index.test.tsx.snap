// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast should renders correctly 1`] = `null`;
